docs/fundamentals/document/document-api.md

```diff
@@ -102,6 +102,37 @@ da.get_attributes('tags__dimensions__height', 'tags__dimensions__weight')
 [[5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0, 5.0], [10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0]]
 ```
 
+
+````{admonition} Note
+:class: caution
+
+As `tags` does not have a fixed schema, it is declared with type `google.protobuf.Struct` in the `DocumentProto`
+protobuf declaration. However, `google.protobuf.Struct` follows the JSON specification and does not 
+differentiate `int` from `float`. **So, data of type `int` in `tags` will be casted to `float` when request is
+sent to executor.**
+
+As a result, users need be explicit and cast the data to the expected type as follows.
+
+```{code-block} python
+---
+emphasize-lines: 7
```

Comment:
You can wrap this inside 2 tabs (Do and Don't) as in here: https://docs.jina.ai/fundamentals/clean-code/
Source code: https://github.com/jina-ai/jina/blob/master/docs/fundamentals/clean-code.md line 13
In the don't tab, you can show what happens if you don't cast

Reply:
Addressed, please see [here](https://github.com/jina-ai/jina/pull/3349/commits/1edd61794053985266a3da7258ec215f96e28458)/